File: leetcode/2. problem/java/674.最长连续递增序列.java
/*
 * @lc app=leetcode.cn id=674 lang=java
 *
 * [674] 最长连续递增序列
 */

// @lc code=start
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        // 贪心
        // return extracted(nums);

        // 动态规划
        // return extracted2(nums);

    }

    private int extracted2(int[] nums) {
        int[] dp = new int[nums.length];
        for (int i = 0; i < dp.length; i++) {
            dp[i] = 1;
        }
        int res = 1;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i + 1] > nums[i]) {
                dp[i + 1] = dp[i] + 1;
            }
            res = Math.max(res, dp[i + 1]);
        }
        return res;
    }

    private int extracted(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int len = 1;
        int start = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] <= nums[i - 1]) {
                start = i;
            }
            len = Math.max(len, i - start + 1);
        }
        return len;
    }
}
// @lc code=end

File: leetcode/2. problem/674.最长连续递增序列.go
/*
 * @lc app=leetcode.cn id=674 lang=golang
 *
 * [674] 最长连续递增序列
 */
package main

// @lc code=start
func findLengthOfLCIS(nums []int) int {
	return newFunction18(nums)

}

// 35/35 cases passed (4 ms)
// Your runtime beats 95.59 % of golang submissions
// Your memory usage beats 28.26 % of golang submissions (4.4 MB)
func newFunction18(nums []int) int {
	ans := 1
	n := len(nums)
	dp := make([]int, n)
	for i := range dp {
		dp[i] = 1
	}
	for i := 1; i < n; i++ {
		if nums[i] > nums[i-1] {
			dp[i] = dp[i-1] + 1
		}
		if dp[i] > ans {
			ans = dp[i]
		}
	}
	return ans
}

// @lc code=end
